Dockerfile

```diff
@@ -2,32 +2,26 @@ FROM node:14.16.1 AS make-bundle
 ENV WEB_APP=/home/app/webapp
 WORKDIR $WEB_APP
 COPY . $WEB_APP
-RUN npm install 
-RUN npm run webpack 
+RUN npm install && \
+    npm run webpack 
+
+
 
-# base image  
 FROM python:3.7  
-# setup environment variable  
 ENV WEB_APP=/home/app/webapp  
-
-# set work directory  
 RUN mkdir -p $WEB_APP  
-
-# where your code lives  
 WORKDIR $WEB_APP  
-
-# set environment variables  
-ENV PYTHONDONTWRITEBYTECODE 1
-ENV PYTHONUNBUFFERED 1  
-# install dependencies  
+ENV PYTHONDONTWRITEBYTECODE 1 \
+    PYTHONUNBUFFERED 1  
 RUN pip install --upgrade pip  
-# copy whole project to your docker home directory. 
-# COPY . $DockerHOME
-COPY --from=make-bundle /home/app/webapp $WEB_APP 
-# run this command to install all dependencies  
-RUN pip install -r requirements.txt  
-# port where the Django app runs  
+COPY --from=make-bundle /home/app/webapp/ $WEB_APP 
+RUN pip install -r requirements.txt && \
+    rm -rf node_modules && \
```

Comment:
Few things:
Deleting something that exists in the layer below doesnt reduce image size since it will forever exist in the layer below ..
Node modules wasnt checked in to git
It was created in the previous build image and u only copied the bundle from there.